### DISCUSSION TITLE
Modélisation des données des contributions

### DISCUSSION THREAD

**Discussion by m-maillot** [SocialGouv/cdtn-admin#1009] [General]
## Contexte

La données est formatées en json au sein de la table `documents`. L'ensemble des contributions sont transformés en JSON à partir de la données en base de données. 

Il y a 2 types de documents générés : réponse générique et réponse conventionnelle.

### Réponse générique

La structure est la suivante :  
* description
* index
* answers
  * generic (contient les infos génériques de la réponse)
  * conventions (contient une liste des CCs avec la réponse associée à chaque CC)

### Réponse conventionnelle

La structure est la suivante : 
* description
* index
* split (à true)
* answers
  * generic (contient les infos génériques de la réponse)
  * conventionAnswer (contient les infos conventionnelles de la réponse)

## Problématique

On dispose aujourd'hui d'une structure différente entre une réponse conventionnelle et générique. Ce qui s'explique par le format utilisé aujourd'hui. Par contre, ce format change et on va disposer aujourd'hui d'une seule réponse pour le conventionnel.

Il serait intéressant de revoir le format pour générer un JSON en phase avec les évolutions.

**Comment by m-maillot** [comment]
## Solution 1 : Laisser comme tel

On génère le même json qu'actuellement. 

Lors de la publication d'une réponse générique, on va récupérer l'ensemble des réponses conventionnelles et alimenter le json avec le contenu de la générique et l'ensemble des réponses conventionnelles.

Lors de la publication d'une réponse conventionnelle, on récupère le contenu de la générique associé et on retourne les deux contenus. On peut voir ici pour retourner null dans la réponse générique qui ne sera pas utilisée.

**Avantages**
 * Pas de modifications sur l'UI actuelle
 * Migration plus simple
 
**Inconvénients**
 * Format complexe qui n'est pas en corrélation avec le contenu affiché
 * Comment identifier le nouveau contenu de l'ancien contenu ?

**Reply by carolineBda** [reply]
"Pas de modifications sur l'UI actuelle" => oui mais nous allons réécrire cette partie quoi qu'il arrive

**Comment by m-maillot** [comment]
## Solution 2 : Revoir le format JSON

Le contenu du JSON serait identique que ce soit pour une réponse générique et une réponse conventionnelle. 

Voici une proposition du contenu du JSON : 

 * index
 * agreement (`undefined` si c'est une réponse générique)
   * idcc
   * shortTitle 
 * answer
   * markdown
   * references 

Cela demande un changement sur l'UI à identifier.

**Reply by carolineBda** [reply]
je dirai que le champs answer n'est plus utile on pourrait avoir markdow et references directement à la racine

**Reply by carolineBda** [reply]
Par contre nous avons toujours besoin de savoir pour une contributions données quand on est sur la généric et qu'on selectionne une CC si cette dernière est traité ou non. Si on part sur cette solution, nous pourrions faire cela avec un appel à l'API ?
